tests/test_integration_http.py

```diff
@@ -650,13 +660,27 @@ def test_capture_proxy(cmake, httpserver, run_args, proxy_status):
         # make sure we are isolated from previous runs
         shutil.rmtree(tmp_path / ".sentry-native", ignore_errors=True)
 
-        httpserver.expect_request("/api/123456/envelope/").respond_with_data("OK")
+        # if proxy_from_env is set, set the proxy environment variables
+        if proxy_from_env == ["proxy-from-env"]:
+            auth = ""
+            if proxy_auth == ["on"]:
+                auth = "user:password@"
+            if run_args == ["http-proxy"]:
+                os.environ["http_proxy"] = f"http://{auth}localhost:8080"
+            elif run_args == ["socks5-proxy"]:
+                os.environ["http_proxy"] = f"socks5://{auth}localhost:1080"
 
+        httpserver.expect_request("/api/123456/envelope/").respond_with_data("OK")
+        current_run_arg = run_args[0]
+        if proxy_auth == ["on"]:
+            current_run_arg += "-auth"
+        if proxy_from_env == ["proxy-from-env"]:
+            current_run_arg = "proxy-from-env"  # overwrite args if proxy-from-env is set (e.g. don't manually set)
```

Comment:
In cURL transport we don't even need to call our proxy-from-env setter, since it will attempt to read the environment variable automatically https://everything.curl.dev/usingcurl/proxies/env.html . 
This kind of makes this feature only really useful for Windows.

Reply:
Is this also true for the library (i.e., `libcurl`)? There are a couple of such features where the library only provides the mechanism, but the reading from env is then implemented in the application (i.e., `curl`).

Reply:
Seems like it is https://curl.se/libcurl/c/libcurl-env.html